src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,76 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private static Map<Class<?>, Object> instances;
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+
+        Class<?> implClazz = getImpl(interfaceClazz);
+        Object clazzImplementationObject = createNewInstance(implClazz);
+        if (!implClazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("Annotation @Component is not present in the class: "
+                    + implClazz.getName());
+        }
```

Comment:
I think we should first check class for Component annotation and then create it's instance